crates/codegen/syntax_templates/src/shared/parse_error.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::Range;
+
+use super::{kinds::*, parse_output::ParseError};
+
+impl ParseError {
+    #[allow(dead_code)]
+    pub(crate) fn new_at_position(
+        utf8_position: usize,
+        tokens_that_would_have_allowed_more_progress: Vec<TokenKind>,
```

Comment:
Not sure I understand why it is diferent than `expected_tokens`. Is it used differently?

```suggestion
        expected_tokens: Vec<TokenKind>,
```

Reply:
Changed

Reply:
I still see 146 instances of `tokens_that_would_have_allowed_more_progress`. I think we missed  a few.

Reply:
Actually, `expected_tokens` should be changed, I missed this. `expected_tokens` is IMO not sufficiently expressive. Having a meaningful name really made this easier to think about

Reply:
I'm fine with using the other name everywhere instead. Just the two names confused me, as I was tracing when does it carry different values.